.github/CONTRIBUTING.md

```diff
@@ -16,41 +16,68 @@ There are 3 core contribution pillars:
+
+Once the goal is clear, you must identify what stops us from achieving it. Anything that is stopping us - is a “Problem”. A typical question to ask is: "why this Goal is not achieved yet and what is the Problem?".
 
 Sometimes, a Goal already has a few identified problems, but it’s not always the case.
 
 > [!NOTE]
 > Once a Problems is identified, we report it as a [GitHub Issue](https://docs.github.com/en/issues) with the following following naming pattern: `Problem: [statement]`. We’re counting on our contributors to identify problems.
 
 ### Solution
+
 The third pillar of successful contribution is the Solution.
 
 Different problems may require different set of skills.  
 Whether it’s code, design, or marketing material, we expect a lean and clean solution from the contributor.
 
 > [!NOTE]
-> If the source-code modification is required, the solution is presented as [GitHub PR (Pull Request)](https://docs.github.com/en/pull-requests). PRs must comply with PR Requirements.
+> Code and design solution is presented as [GitHub PR (Pull Request)](https://docs.github.com/en/pull-requests). All PRs, whether for source-code or design changes, must comply with our PR Requirements.
+> 
 
 #### PR Requirements
 
 > [!WARNING]
-> PRs that do not correspond to the following criterias are usually rejected.
+> PRs that do not correspond to the following criteria are usually rejected.
 
 1. **Small size**. [Keep PRs small](https://artsy.github.io/blog/2021/03/09/strategies-for-small-focused-pull-requests/) and completable within 3-4 hours. If the solution requires more time, then decompose is into smaller PRs.
-1. **Clear naming**. When creating PRs and commits follow [Conventional Commit](https://www.conventionalcommits.org) guidelines. Keep it [clean and simple](https://pulsar.apache.org/contribute/develop-semantic-title/#how-to-write-good-pr-titles).
-1. **Stalled PR**. We usually reject and close PRs which do not have ativity for the last 24 hours. Unless there is a clear comment in thas PR explaining a reason why that PR is stalled.
+2. **Clear naming**. When creating PRs and commits follow [Conventional Commit](https://www.conventionalcommits.org) guidelines. Keep it [clean and simple](https://pulsar.apache.org/contribute/develop-semantic-title/#how-to-write-good-pr-titles).
+3. **Stalled PR**. We usually reject and close PRs which do not have activity for the last 24 hours. Unless there is a clear comment in that PR explaining a reason why that PR is stalled.
+4. **Link PR to Issue**. When creating a PR, make sure to link it to the corresponding problem/issue. Follow the instructions [here](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).
+5. **Report your time**. Make sure to report the time you spent on the PR.
```

Comment:
please use the notation:
```
1. {text}
1. {text}
1. {text}
```

because when you want to place something in the middle, of the list 
```
1.
2.
3.
<<<- here
4.
....
20.
```

you don't need to re-number all the subsequent items.